go-controller/pkg/node/ovn_test.go

```diff
@@ -13,22 +13,18 @@ import (
 	"k8s.io/client-go/kubernetes/fake"
 	"k8s.io/client-go/tools/record"
 
-	egressfirewallfake "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressfirewall/v1/apis/clientset/versioned/fake"
 	apiextensionsfake "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/fake"
 )
 
 var fakeNodeName = "node"
 
 type FakeOVNNode struct {
-	node               *OvnNode
-	watcher            *factory.WatchFactory
-	stopChan           chan struct{}
-	recorder           *record.FakeRecorder
-	fakeClient         *fake.Clientset
-	fakeEgressIPClient *egressipfake.Clientset
-	fakeEgressClient   *egressfirewallfake.Clientset
-	fakeCRDClient      *apiextensionsfake.Clientset
-	fakeExec           *ovntest.FakeExec
+	node       *OvnNode
+	watcher    *factory.WatchFactory
+	stopChan   chan struct{}
+	recorder   *record.FakeRecorder
+	fakeClient *util.OVNClientset
```

Comment:
maybe instead of `fakeClient *util.OVNClientset` just embed the struct? ie,

    type FakeOVNNode struct {
            util.OVNClientset

            node *OvnNode
            ...

then instead of `fakeOvnNode.fakeClient.EgressIPClient...` you'd only need `fakeOvnNode.EgressIPClient...`

Reply:
That won't work, _I think_. I need to access its `fakeClient` field and pass it to the watch factory here: https://github.com/ovn-org/ovn-kubernetes/blob/fa5d4a031d66385b968f8ecbb9012b6d002d1242/go-controller/pkg/node/ovn_test.go#L74 and I don't think there's a way to do that with an embedded field.